### ISSUE TITLE
`BatchNorm` handling for rare node types

### ISSUE THREAD

**Issue by Padarn** [pyg-team/pytorch_geometric#5529]
### 🚀 The feature, motivation and pitch

This feature is intended to support Hetrogenous `BatchNorm` when there are rare node types.

Consider the network

```
class Encoder(torch.nn.Module):
    def __init__(self, ...):
         self.conv = ...
         self.norm = BatchNorm(..)

    def forward(self, x, edge_index):
        x = self.conv(x, edge_index)
        return self.norm(x)


class Predictor(torch.nn.Module):
    def __init__(self, hidden_channels, out_channels, metadata):
        super().__init__()
        self.encoder = to_hetero(Encoder(), metadata=metadata, aggr='sum')

    def forward(self, x_dict, edge_index_dict):
        out = self.encoder(x_dict, edge_index_dict)
        return out
```

The batch size within the `Encoder` will vary based on the number of nodes of that type in the given batch/graph. This will fail when a certain node type only has one example in that batch.

The idea here would be to provide some mechanism to handle this, as currently, during training this will throw an error due to `torch.nn.BatchNorm1D` checking to avoid division by zero.

### Alternatives

Alternative would be to not support  this. Or someone already having a suggestion for how to solve it.

### Additional context

I have not provided a specific implementation idea here as I'd like to first get some feedback on if this make sense or not.

**Comment by rusty1s** [comment]
This is a good idea. Something like this should work:
```python
if x.size(0) <= 1:
    training = layer.training
    layer.eval()
    x = layer(x)
    layer.train(training)

**Comment by Padarn** [comment]
I'll make a PR then to follow up. I was wondering if we should still update the running mean, but perhaps its overly complicated.
